cmd/asset-syncer/utils_test.go

```diff
@@ -707,3 +707,70 @@ func Test_fetchAndImportFiles(t *testing.T) {
 		assert.NoErr(t, err)
 	})
 }
+
+type goodChecksumHTTPClient struct{}
+
+const tags = `{"name":"test/apache","tags":["7.5.1","8.1.1"]}`
+
+func (h *goodChecksumHTTPClient) Do(req *http.Request) (*http.Response, error) {
+	w := httptest.NewRecorder()
+	// Don't accept trailing slashes
+	if strings.HasPrefix(req.URL.Path, "//") {
+		w.WriteHeader(500)
+	}
+
+	w.Write([]byte(tags))
+	return w.Result(), nil
+}
+
+type authenticatedChecksumHTTPClient struct{}
+
+func (h *authenticatedChecksumHTTPClient) Do(req *http.Request) (*http.Response, error) {
+	w := httptest.NewRecorder()
+
+	// Ensure we're sending the right Authorization header
+	if !strings.Contains(req.Header.Get("Authorization"), "Bearer ThisSecretAccessTokenAuthenticatesTheClient") {
+		w.WriteHeader(500)
```

Comment:
Fine as is, but given these functions (`Do`) are test only, if you pass a `t *testing.T` to the function you can `t.Fail()` here rather than having to return an http error and interpret that in the code.

Reply:
well, this is also testing that getting an error code in the response is translated to a function error.